pkg/alibabacloud/metadata/metadata.go

```diff
@@ -67,10 +69,26 @@ func getMetadata(ctx context.Context, path string) (string, error) {
 	}
 
 	defer resp.Body.Close()
-	respBytes, err := io.ReadAll(resp.Body)
+
+	reader := bufio.NewReaderSize(resp.Body, bufferSize)
+	var respBytes []byte
+	if resp.ContentLength == -1 {
+		respBytes = make([]byte, bufferSize)
+	} else {
+		respBytes = make([]byte, resp.ContentLength)
+	}
+	n, err := reader.Read(respBytes)
+
 	if err != nil {
 		return "", err
 	}
+	if n == bufferSize {
+		if _, err := reader.ReadByte(); err == nil {
+			log.Error("Buffer size too small. Please report to developers")
+			return "", io.ErrShortBuffer
+		}
```

Comment:
Why are we checking for nil error? I don't think I'm understanding

Reply:
My initial implementation was wrong, I assumed that buffered reader would return error if the size of buffer would be exceeded which ended up not being the case.

`ReadByte` returns error if there is no new byte to read. `err == nil` means that there are more bytes to read, which means that the response is bigger than 1MB, so we log an error and return.